### PR TITLE
feat: add `all` flag to `pop clean` 

### LINKED ISSUE #225: feat: add clean all option in the `pop clean` command
With the new command `pop clean` we let the user pick what binaries to remove. Add an option in the `multiselect` to remove all the content or introduce `--all` flag.

### PR DESCRIPTION
**Pull request by bolajahmad**
Closes #225 !

Adds a subcommand to `pop clean cache` to allow cleaning all cache artifacts at once. Useful as a shorthand!

### CODE REVIEW

File: crates/pop-cli/src/commands/clean.rs
Comment (brunopgalvao) [inline crates/pop-cli/src/commands/clean.rs:79]:
```suggestion
				"Would you like to cleanup all cached artifacts...\n {list}"))
```

**Review by brunopgalvao — COMMENTED** [review]
```
   Pop CLI : Remove cached artifacts
│
⚙  ℹ️ The cache is located at /Users/bruno/Library/Caches/pop
│  
◆  Would you like to cleanup all cache artifacts...
│   > artifacts : 0MiB, bin : 0MiB, polkadot-chain-spec-generator-v1.2.7 : 25MiB, polkadot-execute-worker : 0MiB, polkadot-execute-worker-v1.11.0 : 13MiB, polkadot-execute-worker-v1.13.0 : 11MiB, polkadot-execute-worker-v1.8.0 : 14MiB, polkadot-parachain-v1.13.0 : 157MiB, polkadot-parachain-v1.8.0 : 168MiB, polkadot-prepare-worker : 0MiB, polkadot-prepare-worker-v1.11.0 : 13MiB, polkadot-prepare-worker-v1.13.0 : 12MiB, polkadot-prepare-worker-v1.8.0 : 13MiB, polkadot-v1.11.0 : 118MiB, polkadot-v1.13.0 : 92MiB, polkadot-v1.8.0 : 102MiB, pop-node : 142MiB, pop-node-v0.1.0-alpha2 : 98MiB
```
It would be nice for the output to be a formatted list.

**Comment by brunopgalvao** [comment]
If I select 'No` I get a "Failed to clean cache`:
```
│  
◇  Would you like to cleanup all cache artifacts...
│   > artifacts : 0MiB, bin : 0MiB, polkadot-chain-spec-generator-v1.2.7 : 25MiB, polkadot-execute-worker : 0MiB, polkadot-execute-worker-v1.11.0 : 13MiB, polkadot-execute-worker-v1.13.0 : 11MiB, polkadot-execute-worker-v1.8.0 : 14MiB, polkadot-parachain-v1.13.0 : 157MiB, polkadot-parachain-v1.8.0 : 168MiB, polkadot-prepare-worker : 0MiB, polkadot-prepare-worker-v1.11.0 : 13MiB, polkadot-prepare-worker-v1.13.0 : 12MiB, polkadot-prepare-worker-v1.8.0 : 13MiB, polkadot-v1.11.0 : 118MiB, polkadot-v1.13.0 : 92MiB, polkadot-v1.8.0 : 102MiB, pop-node : 142MiB, pop-node-v0.1.0-alpha2 : 98MiB
│  No 
│
└  Failed to clean cache
```

File: crates/pop-cli/src/commands/clean.rs
Comment (AlexD10S) [inline crates/pop-cli/src/commands/clean.rs:43]: To maintain consistency with the other commands that only have `execute(self)`, add the add flag `all` in the `CleanCacheCommand` struct and then pass it in the `commands/mod.rs` instead of adding it here.
```
Self::Clean(args) => match args.command {
  clean::Command::Cache(cmd_args) => {
	  // Initialize command and execute
	  clean::CleanCacheCommand { cli: &mut Cli, cache: cache()?, all: cmd_args.all }
		  .execute()
		  .map(|_| Value::Null)
  },
  },
```
So you can use it inside `execute` with `self.all`.

File: crates/pop-cli/src/commands/clean.rs
Comment (AlexD10S) [inline crates/pop-cli/src/commands/clean.rs:64]: You can remove this comment, is clear enough

File: crates/pop-cli/src/commands/clean.rs
Comment (AlexD10S) [inline crates/pop-cli/src/commands/clean.rs:89]: This message appears n times, just move the message after the loop:
`ℹ️ 5 artifacts removed`

**Review by AlexD10S — CHANGES_REQUESTED** [review]
Great job! 
I left some comments to address and would love to see a test of the new functionality before merge it.

If I have a folder in the cache I am getting an error: `Error: Operation not permitted (os error 1)`, but that is out of scope of this PR, will be fixed in: https://github.com/r0gue-io/pop-cli/issues/224

**Comment by bolajahmad** [comment]
> If I select 'No` I get a "Failed to clean cache`:

Is there an alternative behavior you expect, or clearer message? The prompt asks for final confirmation to clear the cache, selecting no means nothing was cleared.

File: crates/pop-cli/src/commands/clean.rs
Comment (bolajahmad) [reply crates/pop-cli/src/commands/clean.rs:43]: I see! That definitely makes sense to do.

File: crates/pop-cli/src/commands/clean.rs
Comment (AlexD10S) [inline crates/pop-cli/src/commands/clean.rs:10]: Remove this. That's why the test is waiting for confirmation.
Use `self.cli.confirm` and `self.cli.outro_cancel `instead.

File: crates/pop-cli/src/commands/clean.rs
Comment (AlexD10S) [inline crates/pop-cli/src/commands/clean.rs:285]: Nice test. One fixed the comment above with the imports, you need to replicate the same message as above to assert the message is the same.

**Review by AlexD10S — CHANGES_REQUESTED** [review]

**Comment by AlexD10S** [comment]
CI is failing when running cargo fmt. Run `cargo fmt --all` to fix it

**Comment by bolajahmad** [comment]
> CI is failing when running cargo fmt. Run `cargo fmt --all` to fix it

All formatting is done and changes are completed too. Let me know if I missed out on any changes.

**Comment by AlexD10S** [comment]
> > CI is failing when running cargo fmt. Run `cargo fmt --all` to fix it
> 
> All formatting is done and changes are completed too. Let me know if I missed out on any changes.

Looking good! One last thing, the CI is failing in `cargo deny`. To test it locally you can see why running `cargo deny check`. To fix it, simple merge main branch, where `openssl` library has been upgraded.

**Review by AlexD10S — APPROVED** [review]

**Comment by brunopgalvao** [comment]
> > If I select 'No` I get a "Failed to clean cache`:
> 
> Is there an alternative behavior you expect, or clearer message? The prompt asks for final confirmation to clear the cache, selecting no means nothing was cleared.

I would say something like: "Aborted" or "Cache not cleaned", or simply exit out without a message. "Failed" implies that there was an attempt to clean the cache and it failed. When in reality the user consciously selected "No".

**Comment by bolajahmad** [comment]
> > > If I select 'No` I get a "Failed to clean cache`:
> > 
> > 
> > Is there an alternative behavior you expect, or clearer message? The prompt asks for final confirmation to clear the cache, selecting no means nothing was cleared.
> 
> I would say something like: "Aborted" or "Cache not cleaned", or simply exit out without a message. "Failed" implies that there was an attempt to clean the cache and it failed. When in reality the user consciously selected "No".

That makes sense. I will use the same message returned for selective cleaning. _**"No artifacts removed"**_

**Review by AlexD10S — APPROVED** [review]
Looks good now! Thanks again for contributing to `pop-cli`

**Comment by evilrobot-01** [comment]
Awesome job, especially the test coverage! 🔥

**Pull request merged**
